packages/api/src/utils/parser.ts

```diff
@@ -621,3 +621,18 @@ export const parseEmailAddress = (from: string): addressparser.EmailAddress => {
   }
   return { name: '', address: from }
 }
+
+export const fetchFavicon = async (
+  url: string
+): Promise<string | undefined> => {
+  try {
+    // get the correct url if it's a redirect
+    const response = await axios.head(url)
+    const realUrl = response.request.res.responseUrl
+    const domain = new URL(realUrl).hostname
+    return `https://api.faviconkit.com/${domain}/32`
```

Comment:
NIce. I think they might have rate limiting but lets see. Can also sign up for their service hopefully.

Reply:
Yes and if it doesn't work well, we can always fetch the HTML content from the domain website and parse the `metadata` from the HTML